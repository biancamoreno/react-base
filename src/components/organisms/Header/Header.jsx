import React, { Component } from 'react';
import PropTypes from "prop-types";
import ImputSearch from '../../molecules/ImputSearch/ImputSearch.jsx'
import './Header.scss';

class Header extends Component {
    constructor(props) {
        super(props)
    }
    render() {
        const { handleChange, handleSearch } = this.props;
        return (
            <header className="header">
                <div className="header__wrapper">
                    <div className="header__logo">
                        <img src="../src/assets/images/Logo_ML.png" />
                    </div>
                    <div className="header__imput">
                        <ImputSearch
                            type='text'
                            placeholder='Nunca deixe de buscar'
                            handleChange={handleChange}
                            handleSearch={handleSearch} />
                    </div>
                </div>
            </header>
        )
    }
}

Header.propTypes = {
    handleChange: PropTypes.func.isRequired,
    handleSearch: PropTypes.func.isRequired
}

export default Header;