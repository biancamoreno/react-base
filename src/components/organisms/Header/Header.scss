@import '../../../assets/variables/variables.scss';

.header {
    display: flex;
    justify-content: center;
    background-color: $yellow;
    padding: 8px 30px;
}