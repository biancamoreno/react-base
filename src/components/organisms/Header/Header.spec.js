import React from 'react';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header test', () => {
    it('should render correctly with props', () => {
        const props = {
            handleChange: jest.fn(),
            handleSearch:jest.fn()
        }

        const wrapper = shallow(<Header {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

});