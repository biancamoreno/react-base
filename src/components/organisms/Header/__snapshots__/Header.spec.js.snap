// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test should render correctly with props 1`] = `
<header
  className="header"
>
  <div
    className="header__wrapper"
  >
    <div
      className="header__logo"
    >
      <img
        src="../src/assets/images/Logo_ML.png"
      />
    </div>
    <div
      className="header__imput"
    >
      <ImputSearch
        handleChange={[MockFunction]}
        handleSearch={[MockFunction]}
        placeholder="Nunca deixe de buscar"
        type="text"
      />
    </div>
  </div>
</header>
`;
